File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.cs
public class Solution {
    public int FindMaxConsecutiveOnes(int[] nums) {
        int right=0, left=0, ret=0, zeroCnt=0;
        
        while(right < nums.Length){
            if(nums[right] == 0){
                zeroCnt++;
                
                while(zeroCnt > 1){
                    if(nums[left] == 0){
                        zeroCnt--;                        
                    }
                    left++;
                }
            }
            right++;
            ret = Math.Max(ret, right-left);
        }
        return ret;
    }
}

File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.ts
function findMaxConsecutiveOnes(nums: number[]): number {
    let l: number = 0;
    let r: number = 0;
    let ret: number=0;
    let zeroCnt: number = 0;
    
    while(r < nums.length){
        if(nums[r] === 0){
            zeroCnt++;
            while(zeroCnt > 1){
                if(nums[l] === 0){
                    zeroCnt--;
                }
                l++;
            }            
        }        
        
        r++;
        ret = Math.max(ret, r-l);
    }
    
    return ret;
};